### PR TITLE
DEV: Re-add polymorphic bookmark columns

### PR DESCRIPTION
**Pull request by martin-brennan**
This commit is a redo of 2f1ddadff7dd47f824070c8a3f633f00a27aacde
which we reverted because it blew up an internal CI check. I looked
into it, and it happened because the old migration to add the bookmark
columns still existed, and those columns were dropped in a post migrate,
so the two migrations to add the columns were conflicting before
the post migrate was run.

------

This commit only includes the creation of the new columns and index,
and does not add any triggers, backfilling, or new data.

A backfill will be done in the final PR when we switch this over.
Intermediate PRs will look something like this:

1. Add an experimental site setting for using polymorphic bookmarks,
and make sure in the places where bookmarks are created or updated
we fill in the columns. This setting will be used in subsequent
PRs as well.
2. Listing and searching bookmarks based on polymorphic associations
Creating post and topic bookmarks using polymorphic associations,
and changing special for_topic logic to just rely on the Topic
bookmarkable_type
3. Querying bookmark reminders based on polymorphic associations
4. Make sure various other areas like importers, bookmark guardian,
and others all rely on the associations
5. Prepare plugins that rely on the Bookmark model to use polymorphic
associations

The final core PR will remove all the setting gates and switch over
to using the polymorphic associations, backfill the bookmarks
table columns, and ignore the old post_id and for_topic colummns.
Then it will just be a matter of dropping the old columns down the
line.

### CODE REVIEW

File: db/migrate/20220322024216_add_bookmark_polymorphic_columns_again.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # frozen_string_literal: true
 
-class AddBookmarkPolymorphicColumns < ActiveRecord::Migration[6.1]
+class AddBookmarkPolymorphicColumnsAgain < ActiveRecord::Migration[6.1]
```
Comment (martin-brennan) [inline db/migrate/20220322024216_add_bookmark_polymorphic_columns_again.rb:3]: Kind of weird that git thinks this is a rename, but OK I guess?

Comment (tgxworld) [inline db/migrate/20220322024216_add_bookmark_polymorphic_columns_again.rb:3]: I'm not sure we need `again` here. Now that we have deleted the old migrations, technically we're only adding it once.

Comment (martin-brennan) [reply db/migrate/20220322024216_add_bookmark_polymorphic_columns_again.rb:3]: Yah good point, will just rename

Comment (martin-brennan) [reply db/migrate/20220322024216_add_bookmark_polymorphic_columns_again.rb:3]: Done

**Review by tgxworld — APPROVED** [review]

**Pull request merged**